documentation/docs/extensibility.md

```diff
@@ -1,86 +1,199 @@
+
+  //access config
+  echo "Current stage config: ${params.config}"
+
+  //execute original stage as defined in the template
+  params.originalStage()
+
+  //access overall pipeline script object
+  echo "Branch: ${params.script.commonPipelineEnvironment.gitBranch}"
+
+  echo "End - Extension for stage: ${params.stageName}"
+}
+return this
+```
+
+!!! note "`return this`"
+    Don't forget the `return this` which is required at the end of _all_ extension scripts.
+    This is due to how Groovy loads scripts internally.
 
 !!! note "Init stage cannot be extended"
-    Please note, the `Init` stage among other things also checks out your current repository.<br />Thus it is not possible to use extensions on this stage.
+    Please note, the `Init` stage among other things also checks out your current repository including your extensions.<br />
+    Thus it is not possible to use extensions on this stage.
+
+### Practical example
+
+For a more practical example, you can use extensions in SAP Cloud SDK Pipeline to add custom linters to the pipeline.
+
+A linter is a tool that can check the source code for certain stylistic criteria, and many teams chose to use a linter to ensure a common programming style.
+
+As an example, if you want to use [Checkstyle](https://checkstyle.sourceforge.io/) in your codebase, you might use an extension similar to this one in a file called `.pipeline/extensions/lint.groovy` in your project:
+
+```groovy
+def call(Map parameters) {
+
+    parameters.originalStage() // Runs the built in linters
+
+    mavenExecute(
+        script: parameters.script,
+        flags: '--batch-mode',
+        pomPath: 'application/pom.xml',
+        m2Path: s4SdkGlobals.m2Directory,
+        goals: 'checkstyle:checkstyle',
+    )
+
+    recordIssues blameDisabled: true,
+        enabledForFailure: true,
+        aggregatingResults: false,
+        tool: checkStyle()
+}
+
+return this
+```
+
+This example can be adapted for other linters of your choice.
+
+## 2) Modified ready-made pipeline
+
+This option describes how you can copy and paste one of the centrally maintained pipelines to make changes not possible otherwise.
+For example, you can't change the order of stages, change which stages run in parallel or add new stages to a centrally maintained pipeline.
+This might be done for an individual project (in the `Jenkinsfile`), or in a separate git repository so it can be used for multiple projects.
+
+### Single project
+
+The default `Jenkinsfile` of centrally maintained pipelines does nothing except for loading the pipeline and running it.
+This is convenient, but limits which aspects of the pipeline are modifiable.
+
+If you have one project using the pipeline, the easiest way to do this modification is to copy the pipeline into your `Jenkinsfile`.
+
+The basic structure of your `Jenkinsfile` should be like this:
+
+```groovy
+@Library(/* Shared library definition, see ¹ */) _
```

Comment:
How about instead of this 1 symbol used twice below you write:
"See hint below" and add a note box below with !!!

Reply:
thanks, did that